### ISSUE TITLE
Wish: Upgrade react-popper to version 2

### ISSUE THREAD

**Issue by apepper** [reactstrap/reactstrap#1972]
__tl;dr;__ Please upgrade `react-popper` from version 1 to version 2, since `popper.js` is deprecated.

### Details

Currently `reactstrap` relies on `react-popper` version 1:

https://github.com/reactstrap/reactstrap/blob/c6a790c7f4bdbfac0540d53896834638edf9c793/package.json#L260

`react-popper` version 1 itself relies on `popper.js` version 1:

https://github.com/popperjs/react-popper/blob/b451e968fc28b9ab59775742d82c219e92d11f20/package.json#L66

`popper.js` in the current form (version 1) is now deprecated. https://www.npmjs.com/package/popper.js reads:

> *This package has been deprecated*
> Author message:
> 
> You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1

You can also see this in your terminal, if you do a fresh `npm install`:

```
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
```

It would be awesome, if `reactstrap` upgrades to `react-popper` version 2, so that we no longer see the deprecation warning for `popper.js` version 1.

**Comment by malihassan20** [comment]
@eddywashere can I take this process?

**Comment by kyletsang** [comment]
This is a breaking change.

Popper modifiers have also been changed in v2, so this would need to be adjusted as well.  It's worth noting that bootstrap 4 still uses popper v1, so I'm not sure I see the benefit to moving to v2 at this point?

**Comment by krnlde** [comment]
Bootstrap 5 is final and uses v2. See the benefit now?

**Comment by sbaechler** [comment]
Would it be possible to backport #2104 to V8? 

Even though it is a breaking change it is often not possible for projects to update from Bootstrap 4 to Bootstrap 5. 

Popper 1 makes testing with JSDOM quite difficult and sometimes even impossible.

**Comment by davidacevedo** [comment]
I'm going to close this as reactstrap v9 uses popper v2 with the bootstrap 5 update. I don't see too much of a benefit of backporting popper to V8 as bootstrap 4 still uses popper v1 and popper v2 has some breaking changes.
